host-spec/c04-networking.tm

```diff
@@ -33,36 +33,83 @@
+    and is structured as defined in the libp2p specification
+    (<slink|https://docs.libp2p.io/concepts/peer-id/>).
+  </definition>
+
+  <subsection|Discovery mechanism><label|sect-discovery-mechanism>
 
-  The Polkadot Host uses varies mechanism to find peers within the network,
+  The Polkadot Host uses various mechanisms to find peers within the network,
   to establish and maintain a list of peers and to share that list with other
   peers from the network.
 
-  The Polkadot Host uses various mechanism for peer dicovery.
-
   <\itemize>
     <item>Bootstrap nodes - hard-coded node identities and addresses provided
-    by network configuration itself. Those addresses are selected an updated
-    by the developers of the Polkadot Host. Node addresses should be selected
-    based on a reputation metric, such as reliability and uptime.
+    by the genesis state specification as described in Appendix
+    <reference|sect-genesis-block>.
 
     <item>mDNS - performs a broadcast to the local network. Nodes that might
-    be listing can respond the the broadcast.
-
-    <item>Kademlia requests - Kademlia supports <verbatim|FIND_NODE>
-    requests, where nodes respond with their list of available peers.
+    be listing can respond the the broadcast. The libp2p mDNS specification
+    defines this process in more detail (<slink|><slink|https://github.com/libp2p/specs/blob/master/discovery/mdns.md>).
+
+    <item>Kademlia requests - Kademlia is a distributed hash table for
+    decentralized networks and supports <verbatim|FIND_NODE> requests, where
+    nodes respond with their list of available peers. Kademlia requests must
+    also contain protocol identifiers as described in Section
+    <reference|sect-protocol-identifier>.
   </itemize>
 
+  <subsubsection|Protocol Identifier><label|sect-protocol-identifier>
+
+  Kademlia nodes only communicate with other nodes using the same protocol
+  identifier. The Polkadot network is identified by <verbatim|dot>
+  (respectively <verbatim|ksmcc3> for Kusama).
+
   <subsection|Connection establishment>
 
+  Polkadot nodes connect to peers by establishing a TCP connection. Once
+  established, the node initiates a handshake with the remote peer's
+  encryption layer. An additional layer, know as the multiplexing layer,
+  allows a connection to be split into substreams, either by the local or
+  remote node.
+
+  The Polkadot node supports two kind of substream protocols:
+
+  <\itemize-dot>
+    <item><strong|Request-Response substreams>. After the protocol is
+    negotiated, the initiator sends a single message containing a request.
+    The responder then sends a response, after which the substream is then
+    immediately closed.
+
+    <item><strong|Notification substreams>. After the protocol is negotiated,
+    the initiator sends a single hand shake message. The responder can then
+    either accept or reject the substream. After the substream has been
```

Comment:
Accept by sending back its own length-prefixed handshake, or reject by closing the substream

Reply:
Not sure if it's something to mention, but after having accepted the substream, the remote keep its sending side of the substream open (despite not sending anything anymore), and can later close it in order to ask the initiator to go away.